src/commands/compile.ts

```diff
@@ -413,18 +451,24 @@ export async function namespaceCompile(askFlags = false): Promise<any> {
 function importFiles(files: string[], noCompile = false) {
   return Promise.all<CurrentFile>(
     files.map(
-      throttleRequests((file: string) =>
-        vscode.workspace.fs
-          .readFile(vscode.Uri.file(file))
-          .then((contentBytes) => new TextDecoder().decode(contentBytes))
-          .then((content) => currentFileFromContent(vscode.Uri.file(file), content))
+      throttleRequests((file: string) => {
+        const uri = vscode.Uri.file(file);
+        return vscode.workspace.fs
+          .readFile(uri)
+          .then((contentBytes) => {
+            if (isText(file, Buffer.from(contentBytes))) {
+              return currentFileFromContent(uri, new TextDecoder().decode(contentBytes));
+            } else {
+              return currentFileFromContent(uri, Buffer.from(contentBytes));
+            }
+          })
           .then((curFile) =>
             importFile(curFile).then((data) => {
               outputChannel.appendLine("Imported file: " + curFile.fileName);
               return curFile;
             })
-          )
-      )
+          );
+      })
     )
   ).then(noCompile ? Promise.resolve : compile);
```

Comment:
One other nitpick, the extension tries to compile binary files. Should they be exempted from compilation?

Reply:
@gjsjohnmurray I can implement this if you think it's worth it. If not I can approve and merge the PR.

Reply:
Pushed a change to skip them